Jenkinsfile

```diff
@@ -0,0 +1,223 @@
+      // https://wiki.jenkins-ci.org/display/JENKINS/Email-ext+plugin#Email-extplugin-PipelineExamples
+      // http://stackoverflow.com/a/39535424/14379
+      // IRC: https://issues.jenkins-ci.org/browse/JENKINS-33922
+      // possible alternatives: Slack, HipChat, RocketChat, Telegram?
+      notify.successful()
+    }
+  }
+} catch (e) {
+  notify.failed()
+  throw e
+}
+
+// TODO factor these out into zanata-pipeline-library too
+
+void xvfb(Closure wrapped) {
+  wrap([$class: 'Xvfb']) {
+    wrapped.call()
+  }
+}
+
+void debugChromeDriver() {
+  sh returnStatus: true, script: 'which chromedriver google-chrome'
+  sh returnStatus: true, script: 'ls -l /opt/chromedriver /opt/google/chrome/google-chrome'
+}
+
+void integrationTests(def appserver) {
+  sh "find . -path \"*/${failsafeTestReports}\" -delete"
+
+  try{
+    xvfb {
+      withPorts {
+        // Run the maven build
+        sh """./run-clean.sh ./mvnw -e verify \
+                   --batch-mode \
+                   --settings .travis-settings.xml \
+                   -Danimal.sniffer.skip=true \
+                   -DstaticAnalysis=false \
+                   -Dcheckstyle.skip \
+                   -Dappserver=$appserver \
+                   -Dcargo.debug.jvm.args= \
+                   -DskipUnitTests \
+                   -Dmaven.main.skip \
+                   -Dgwt.compiler.skip \
+                   -Dwebdriver.display=${env.DISPLAY} \
+                   -Dwebdriver.type=chrome \
+                   -Dwebdriver.chrome.driver=/opt/chromedriver \
+                   -DallFuncTests
+          """
+      }
+    }
+  }catch(e){
+    currentBuild.result = 'UNSTABLE'
+    archiveTestFilesIfUnstable()
+    throw e
+  }finally{
+    notify.testResults(appserver.toUpperCase())
+    junit allowEmptyResults: true,
+        keepLongStdio: true,
+        testDataPublishers: [[$class: 'StabilityTestDataPublisher']],
+        testResults: "**/${failsafeTestReports}"
+  }
+}
+
+void withPorts(Closure wrapped) {
+  def ports = sh(script: 'server/etc/scripts/allocate-jboss-ports', returnStdout: true)
+    withEnv(ports.trim().readLines()) {
+      wrapped.call()
+    }
+}
+
+// Modify classnames of tests, to avoid collision between EAP and WildFly test runs.
+// We also use this to distinguish unit tests from integration tests.
+// from https://issues.jenkins-ci.org/browse/JENKINS-27395?focusedCommentId=256459&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-256459
+void setJUnitPrefix(prefix, files) {
+  // add prefix to qualified classname
+  sh "find . -path \"*/${file}\" -exec sed -i \"s/\\(<testcase .*classname=['\\\"]\\)\\([a-z]\\)/\\1${prefix}.\\2/g\" '{}' +"
+}
+
+void archiveTestFilesIfUnstable() {
+  // if tests have failed currentBuild.result will be 'UNSTABLE'
+  if (currentBuild.result != null) {
+    archive(
+        includes: '*/target/**/*.log,*/target/screenshots/**,**/target/site/jacoco/**,**/target/site/xref/**',
```

Comment:
@definite On reflection, I don't think we should archive jacoco results for failed tests.